### PR TITLE
feat: setup prettier, eslint, conventional commits, prevent push to main branch

### PR DESCRIPTION
**Pull request by ahmadshaheer**
Part of https://github.com/SigNoz/engineering-pod/issues/1858

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+pnpm lint-staged
```
Comment (YounixM) [inline .husky/pre-commit:4]: Should we add prettier and typecheck for pre-commit hooks ?

File: package.json [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
+		"dev": "turbo run dev",
+		"lint": "turbo run lint",
+		"clean": "turbo run clean && rm -rf node_modules",
+		"format": "prettier --write \"**/*.{ts,tsx,md}\"",
+		"changeset": "changeset",
+		"version-packages": "changeset version",
+		"update-version": "changeset && changeset version",
+		"release": "turbo run build --filter=docs^... && changeset publish",
+		"prepare": "husky install",
+		"pre-commit": "lint-staged"
```
Comment (YounixM) [inline package.json:14]: Have we setup typecheck and prettier for this repo ?

File: .husky/pre-commit (hunk 1 above)
Comment (ahmadshaheer) [reply .husky/pre-commit:4]: Done

**Review by vikrantgupta25 — APPROVED** [review]
LGTM

**Review by vikrantgupta25 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**